packages/LSOnline/modules/player/playerEvents.js

```diff
@@ -65,10 +55,46 @@ mp.events.add({
   },
 
   playerChat: (player, text) => {
+    if (!player.isLogged) {
+      return player.kick();
+    }
+
     if (player.brutallyWounded) {
       return player.call('actionDone', ['Nie możesz tego teraz użyć!', 'Twoja postać jest nieprzytomna!']);
     }
 
     mp.players.broadcastInRange(player.position, 25, player.dimension, `${player.name} mówi: ${text}`);
   }
 });
+
+const createQuitLabel = (player, exitType) => {
+  if (!player.quitLabel) {
+    player.quitLabel = mp.labels.new(`~HUD_COLOUR_GREYLIGHT~ (( ${player.name} - ${exitType} ))`, new mp.Vector3(player.position.x, player.position.y, player.position.z),
+      {
+        los: true,
+        font: 0,
+        drawDistance: 10,
+        dimension: player.dimension
+      });
+  }
+
+  // This need to be rewrited when we gonna create labels manager.
+  setTimeout(() => {
+    if (player.quitLabel) {
+      player.quitLabel.destroy();
+    }
+  }, 50000);
+};
+
+const prepareBeforeQuit = (player, exitType = false) => {
```

Comment:
Function should be extracted somewhere else

Reply:
look at 53f3925.